src/main/scala/exercises/05_Patterns.scala

```diff
@@ -56,15 +62,23 @@ object PatternMatching {
    *    - anything else, the function result is "Some Scala class"
    */
   def matchOnInputType(in: Any): String = {
-    error("fix me")
+    in match {
+      case a: String => "A string with length " + a.length
+      case b: Int => "A positive integer"
+      case c: Person => "A person with name: " + c.name
+      case d: Seq[Any] if d.length > 10 => "Seq with more than 10 elements"
+      case e: Seq[Any] if e.length > 2 => "first: " + e(0) + ", second: " + e(1) + ", rest: " + e.takeRight(e.length - 2)
+      case null => "A null value"
+      case f => "Some Scala class"
```

Comment:
I also did not see the if operator for pattern matching. It seems like there are a lot of ways to clean up patterns.